template/cordova/Api.js

```diff
@@ -0,0 +1,54 @@
+
+/*jshint node: true*/
+
+var path = require('path');
+
+var buildImpl = require('./lib/build');
+var requirementsImpl = require('./lib/check_reqs');
+
+function PlatformApi () {
+    // Set up basic properties. They probably will be overridden if this API is used by cordova-lib
+    this.root = path.join(__dirname, '..', '..');
+    this.platform = 'windows';
+
+    if (this.constructor.super_){
+        // This should only happen if this class is being instantiated from cordova-lib
+        // In this case the arguments is being passed from cordova-lib as well,
+        // so we don't need to care about whether they're correct ot not
+        this.constructor.super_.apply(this, arguments);
+    }
+}
+
+PlatformApi.prototype.build = function(context) {
+    var buildOptions = context && context.options || [];
+    return buildImpl.run(buildOptions);
+};
+
+PlatformApi.prototype.requirements = function () {
+    return requirementsImpl.check_all();
+};
+
+function PlatformHandler() {
```

Comment:
Do we want to keep PlatformHandler and PlatformApi as 2 separate classes long term?
From the current CLI style workflow, the methods in PlatformHandler seem to be like something private that should be hidden from the user. But actually some of the methods like getWwwDir() might be very useful to the user. An example would be when the web app is very large and copying it all on updateWww become a heavy operation. But if the user knows where to copy the files to, he can fine-tune his workflow to only copy changes which is a very common case when working with gulp/grunt watch.